src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -809,23 +808,34 @@ public IEnumerable<IWithdrawal> GetWithdrawals()
         }
 
         /// <inheritdoc />
-        public List<(Transaction, TransactionData, IWithdrawal)> FindWithdrawalTransactions(uint256 depositId = null)
+        public List<(Transaction, IWithdrawal)> FindWithdrawalTransactions(uint256 depositId = null)
         {
+            // A withdrawal is a transaction that spends funds from the multisig wallet.
             lock (this.lockObject)
             {
-                List<(Transaction, TransactionData, IWithdrawal)> withdrawals = new List<(Transaction, TransactionData, IWithdrawal)>();
+                var withdrawals = new List<(Transaction, IWithdrawal)>();
 
                 foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions)
                 {
-                    Transaction walletTran = transactionData.GetFullTransaction(this.network);
-                    IWithdrawal withdrawal = this.withdrawalExtractor.ExtractWithdrawalFromTransaction(walletTran, transactionData.BlockHash, transactionData.BlockHeight ?? 0);
+                    if (transactionData.SpendingDetails == null)
+                        continue;
+
+                    Transaction walletTran = this.network.CreateTransaction(transactionData.SpendingDetails.Hex);
+
+                    if (withdrawals.Any(w => w.Item1.GetHash() == walletTran.GetHash()))
```

Comment:
Is this implying that there can be duplicates in the multisig wallet?

Reply:
Off the top of my head, you can have multiple inputs in the wallet getting spent together into one transaction, so it would appear in multiple UTXOs spending details. Whether that is something that frequently happens is debatable.

Reply:
Ah so its a scenario where you spend some inputs from a transaction in one situation and then some other inputs in another?

Reply:
yes, I agree with @zeptin. 1 withdrawal - many transaction data (UTXO) being spent.